lib/pf/Authentication/Source/LDAPSource.pm

```diff
@@ -101,35 +106,50 @@ sub authenticate {
   }
 
   my $filter = "($self->{'usernameattribute'}=$username)";
+
+  $before = Time::HiRes::gettimeofday();
   $result = $connection->search(
     base => $self->{'basedn'},
     filter => $filter,
     scope => $self->{'scope'},
     attrs => ['dn']
   );
+  $pf::StatsD::statsd->end("LDAPSource::" . called() . ".search.timing" , $before, 0.25 ); 
+
   if ($result->is_error) {
     $logger->error("[$self->{'id'}] Unable to execute search $filter from $self->{'basedn'} on $LDAPServer:$LDAPServerPort");
+    $pf::StatsD::statsd->end("LDAPSource::" . called() . ".timing" , $start, 0.25 ); 
+    $pf::StatsD::statsd->increment("LDAPSource::" . called() . ".error.count" );
     return ($FALSE, $COMMUNICATION_ERROR_MSG);
   }
 
   if ($result->count == 0) {
     $logger->warn("[$self->{'id'}] No entries found (". $result->count .") with filter $filter from $self->{'basedn'} on $LDAPServer:$LDAPServerPort");
+    $pf::StatsD::statsd->end("LDAPSource::" . called() . ".timing" , $start, 0.25 ); 
+    $pf::StatsD::statsd->increment("LDAPSource::" . called() . ".failure.count" );
     return ($FALSE, $AUTH_FAIL_MSG);
   } elsif ($result->count > 1) {
     $logger->warn("[$self->{'id'}] Unexpected number of entries found (" . $result->count .") with filter $filter from $self->{'basedn'} on $LDAPServer:$LDAPServerPort for source $self->{'id'}");
+    $pf::StatsD::statsd->end("LDAPSource::" . called() . ".timing" , $start, 0.25 ); 
+    $pf::StatsD::statsd->increment("LDAPSource::" . called() . ".failure.count" );
     return ($FALSE, $AUTH_FAIL_MSG);
   }
 
   my $user = $result->entry(0);
 
+  $before = Time::HiRes::gettimeofday();
   $result = $connection->bind($user->dn, password => $password);
+  $pf::StatsD::statsd->end("LDAPSource::" . called() . ".bind.timing" , $before, 0.25 );
 
   if ($result->is_error) {
     $logger->warn("[$self->{'id'}] User " . $user->dn . " cannot bind from $self->{'basedn'} on $LDAPServer:$LDAPServerPort");
+    $pf::StatsD::statsd->end("LDAPSource::" . called() . ".timing" , $start, 0.25 ); 
```

Comment:
Putting the ID in this would make it much clearer.

Ex : 
"LDAPSource::" . $self->id . called() . ".timing"